src/pr-trigger.ts

```diff
@@ -4,175 +4,147 @@ import { queryPRInfo, deriveStateForPR } from "./pr-info";
+                body: `NOOPing because typescript-bot triggered the request`
             };
             return;
         }
-        prNumber = pr.number;
-    }
-
-    if (prNumber === -1) throw new Error(`PR Number was not set from a webhook - ${event} on ${action}`);
 
-    if (prHandlers.has(prNumber)) prHandlers.get(prNumber)(); // cancel older handler for the same pr
-    const aborted = await new Promise(res => {
-        const timeout = setTimeout(() => res(false), 30000);
-        prHandlers.set(prNumber, () => {
-            clearTimeout(timeout);
-            res(true);
-        });
-    });
-    if (aborted) {
-        context.log.info(`Skipped webhook, superseded by a newer one for ${prNumber}`);
-        context.res = {
-            status: 204,
-            body: "NOOPing due to a newer webhook"
-        };
-        return;
-    }
-    prHandlers.delete(prNumber);
-
-    context.log.info(`Getting info for PR ${prNumber} - ${prTitle}`);
-
-    // Generate the info for the PR from scratch
-    const info = await queryPRInfo(prNumber);
-    const prInfo = info.data.repository?.pullRequest;
+        // Allow the bot to run side-effects which are not the 'core' function
+        // of the review cycle, but are related to keeping DT running smoothly
+        if (event.name === "check_suite") {
+            await mergeCodeOwnersOnGreen(event.payload);
+        }
 
-    // If it didn't work, bail early
-    if (!prInfo) {
-        const isIssueNotPR = "issue" in webhook;
-        if (isIssueNotPR) {
+        const pr: { number: number, title?: string } | undefined = prFromEvent();
+        if (!pr) {
+            context.log.info("Skipped webhook, could not find a PR");
             context.res = {
                 status: 204,
-                body: `NOOPing due to ${prNumber} not being a PR`
+                body: "NOOPing due to not finding a PR"
+            };
+            return;
+        }
+
+        prHandlers[pr.number]?.(); // cancel older handler for the same pr
+        const aborted = await new Promise(res => {
+            const timeout = setTimeout(() => res(false), 30000);
+            prHandlers[pr.number] = () => {
+                clearTimeout(timeout);
+                res(true);
             };
-        } else {
-            context.log.error(`No PR with this number exists, (${JSON.stringify(info)})`);
+        });
+        if (aborted) {
+            context.log.info(`Skipped webhook, superseded by a newer one for ${pr.number}`);
             context.res = {
-                status: 422,
-                body: `No PR with this number exists, (${JSON.stringify(info)})`
+                status: 204,
+                body: "NOOPing due to a newer webhook"
             };
+            return;
+        }
+        delete prHandlers[pr.number];
+
+        context.log.info(`Getting info for PR ${pr.number} - ${pr.title || "(title not fetched)"}`);
+
+        // Generate the info for the PR from scratch
```

Comment:
The "from scratch" part of this comment should have been dropped a while ago...

Reply:
Removed. :heavy_check_mark:

Reply:
👍